app/src/main/java/com/github/se/icebreakrr/ui/map/MapView.kt

```diff
@@ -349,3 +436,66 @@ fun MarkerOverlay(position: Offset, text: String) {
             )
       }
 }
+
+@Composable
+fun ProfileModal(
+    profile: Profile?,
+    locationDescription: String?,
+    onDismiss: () -> Unit,
+    onNavigate: (String) -> Unit
+) {
+  if (profile != null) {
+    Box(
+        modifier =
+            Modifier.fillMaxSize().background(Color.Black.copy(alpha = 0.5f)) // Blurred background
+        ) {
+          Card(
+              modifier = Modifier.align(Alignment.Center).padding(16.dp).testTag("profileModal"),
+              shape = RoundedCornerShape(16.dp),
+              onClick = { onNavigate(Screen.OTHER_PROFILE_VIEW + "?userId=${profile.uid}") }) {
+                Column(
+                    modifier = Modifier.padding(16.dp),
+                    horizontalAlignment = Alignment.CenterHorizontally, // Center horizontally
+                    verticalArrangement = Arrangement.spacedBy(16.dp) // Add spacing between items
+                    ) {
+                      AsyncImage(
+                          model = profile.profilePictureUrl,
+                          contentDescription = "profile picture",
+                          modifier =
+                              Modifier.size(IMAGE_SIZE)
+                                  .clip(CircleShape)
+                                  .align(Alignment.CenterHorizontally), // Center the image
+                          placeholder =
+                              painterResource(id = R.drawable.nopp), // Default image during loading
+                          error =
+                              painterResource(id = R.drawable.nopp) // Fallback image if URL fails
+                          )
+                      Text(
+                          text = profile.name,
+                          fontSize = 24.sp,
+                          fontWeight = FontWeight.Bold,
+                          modifier = Modifier.align(Alignment.CenterHorizontally) // Center the text
+                          )
+                      // Display the location description
+                      locationDescription?.let {
+                        Text(
+                            text = it,
+                            fontSize = 14.sp,
+                            fontWeight = FontWeight.Normal,
+                            modifier =
+                                Modifier.align(
+                                    Alignment.CenterHorizontally) // Center the location description
```

Comment:
I suggest you to declare constant `TextStyle` at the beginning of the file and use the `style` parameter of `Text`. This will make the code easier to harmonize later if we want to centralize every textstyles.

Reply:
Okay sure!